### PR TITLE
Skip test if adios2 python module built without MPI.

### PR DESCRIPTION
**Pull request by jhale**
It would also be good to not build against ADIOS2 if it wasn't built against MPI...

### CODE REVIEW

File: python/test/unit/io/test_adios2.py
Comment (drew-parsons) [inline python/test/unit/io/test_adios2.py:294]: Keep in mind `adios2.is_built_with_mpi` is new, i.e. not available in ADIOS2 2.8.1.  2.10.0 has it (released two weeks ago)

File: cpp/CMakeLists.txt
Comment (drew-parsons) [inline cpp/CMakeLists.txt:375]: might want to bump to ADIOS2 2.10.0 for `adios2.is_built_with_mpi` (see below)
I suggest not pushing this into dolfinx 0.8 since ADIOS2 2.10 has only just been released.

File: python/test/unit/io/test_adios2.py
Comment (drew-parsons) [reply python/test/unit/io/test_adios2.py:294]: Alternatively could place the test is a try block and catch the `AttributeError` from adios2 2.8 or 2.9.  Perhaps that could be more helpful for the dolfinx 0.8 release

File: python/test/unit/io/test_adios2.py
Comment (drew-parsons) [inline python/test/unit/io/test_adios2.py:273]: This line looks like a git conflict marker that shouldn't be here (needs reconciliation after merging from HEAD?)

**Review by drew-parsons — CHANGES_REQUESTED** [review]

**Comment by jorgensd** [comment]
@drew-parsons We have decided that this test will only run if the user supplies `adios2>=2.10` as this is not a requirement in the actual code base.

File: cpp/CMakeLists.txt
Comment (jorgensd) [reply cpp/CMakeLists.txt]: We only require this in one test in Python, so we can keep the lower limit (2.8.1) in C++.

**Review by jorgensd — APPROVED** [review]

**Review by drew-parsons — APPROVED** [review]
Makes sense to check `adios2.is_built_with_mpi` if the adios2 serial build can't be used.

File: python/test/unit/io/test_adios2.py
Comment (garth-wells) [inline python/test/unit/io/test_adios2.py:317]: This is overcomplicated. Looking at the ADIOS2 code, the attribute `is_built_with_mpi` is always defined. If it's not in earlier versions, put ADIOS2 version check in a marker for the whole test.

**Review by garth-wells — CHANGES_REQUESTED** [review]

File: python/test/unit/io/test_adios2.py
Comment (jorgensd) [reply python/test/unit/io/test_adios2.py:317]: Version checks are not trivial without external libraries (i think), Ref https://til.simonwillison.net/python/comparing-version-numbers

Comment (drew-parsons) [reply python/test/unit/io/test_adios2.py:317]: True.  I wasn't sure how easy it would be to set up which is why I didn't mention it.
mpi4py uses [versioned conditions](https://github.com/mpi4py/mpi4py/blob/fd1abfeb86305623e4381801f3a99b70f6505e51/test/test_spawn.py#L86) but that's implemented via [custom rules](https://github.com/mpi4py/mpi4py/blob/fd1abfeb86305623e4381801f3a99b70f6505e51/test/mpiunittest.py#L27). Similar h5py has [version conditions](https://github.com/h5py/h5py/blob/862eecefc72eb6941db3b370dff27a6e18bf6197/h5py/tests/test_file.py#L238), but on its own version provided as a tuple.

Comment (drew-parsons) [reply python/test/unit/io/test_adios2.py:317]: That said, it could be simplified with 
```
@pytest.mark.skipif(not hasattr(adios2, "is_built_with_mpi"), "requires adios2>=2.10.0")
```

Comment (jorgensd) [reply python/test/unit/io/test_adios2.py:317]: Thats not quite right as adios2 is only defined through importorskip

Comment (drew-parsons) [reply python/test/unit/io/test_adios2.py:317]: importorskip would already skip the entire test file though.
Then too, importorskip has its own [version parsing](https://docs.pytest.org/en/6.2.x/skipping.html#skipping-on-a-missing-import-dependency)
```
adios2 = pytest.importorskip("adios2", minversion="2.10.0")
```

**Comment by jorgensd** [comment]
@garth-wells are you happy with the simplification? I can't get it to auto-merge without "force-dismissing" your review, which seems rather rude;)

**Review by garth-wells — APPROVED** [review]

**Pull request merged**
